daemon/src/connection.rs

```diff
@@ -116,9 +116,13 @@ impl State {
             return false;
         }
 
+        let heart_beat = self
+            .last_heartbeat()
+            .map(|heart_beat| format!("{heart_beat}"))
```

Comment:
Curious why clippy doesn't flag this as unnecessary format call. You can use to_string here.